tests/report/common/tests.cpp

```diff
@@ -2,15 +2,292 @@
+    }
+    else if (header->report_type == OE_REPORT_TYPE_SGX_LOCAL)
+    {
+        if (collaterals != NULL || collaterals_size > 0)
+        {
+            OE_RAISE_MSG(
+                OE_UNSUPPORTED,
+                "Local reports should not have collaterals.",
+                NULL);
+        }
+
+#ifndef OE_BUILD_ENCLAVE
+        if (enclave == NULL)
+            OE_RAISE(OE_INVALID_PARAMETER);
+
+        OE_CHECK(oe_verify_report(enclave, report, report_size, parsed_report));
+#else
+        OE_CHECK(oe_verify_report(report, report_size, parsed_report));
+#endif
+    }
+    else
+    {
+        OE_RAISE(OE_INVALID_PARAMETER);
+    }
+
+    result = OE_OK;
+done:
+    return result;
+}
+
+/**
+ * Free up any resources allocated by oe_get_collateras()
+ *
+ * @param collaterals_buffer The buffer containing the collaterals.
+ */
+static void oe_free_collaterals(uint8_t* collaterals_buffer)
+{
+    oe_free_collaterals_internal(collaterals_buffer);
+}
+
+static oe_result_t oe_get_quote_validity_with_collaterals(
+    const uint8_t* report,
+    const size_t report_size,
+    const uint8_t* collaterals,
+    size_t collaterals_size,
+    oe_datetime_t* valid_from,
+    oe_datetime_t* valid_until)
+{
+    oe_result_t result = OE_UNEXPECTED;
+    oe_report_t oe_report = {0};
+    oe_report_header_t* header = (oe_report_header_t*)report;
+
+    if (report == NULL || collaterals == NULL || valid_from == NULL ||
+        valid_until == NULL)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    if (report_size == 0 || report_size > OE_MAX_REPORT_SIZE ||
+        collaterals_size == 0)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    // Ensure that the report is parseable before using the header.
+    OE_CHECK(oe_parse_report(report, report_size, &oe_report));
+
+    if (header->report_type == OE_REPORT_TYPE_SGX_REMOTE)
+    {
+#ifndef OE_BUILD_ENCLAVE
+        // Intialize the quote provider if we want to verify a remote quote.
+        // Note that we don't have the OE_USE_LIBSGX guard here since we don't
+        // need the sgx libraries to verify the quote. All we need is the quote
+        // provider.
+        OE_CHECK(oe_initialize_quote_provider());
+#endif
+
+        // Quote attestation can be done entirely on the host side.
+        OE_CHECK(oe_get_quote_validity_with_collaterals_internal(
+            header->report,
+            header->report_size,
+            collaterals,
+            collaterals_size,
+            valid_from,
+            valid_until));
+    }
+    else
+    {
+        OE_RAISE(OE_INVALID_PARAMETER);
+    }
+
+    result = OE_OK;
+done:
+    return result;
+}
+
 #ifdef OE_BUILD_ENCLAVE
 #include <openenclave/corelibc/string.h>
 
 #define GetReport oe_get_report
 #define GetReport_v2 oe_get_report_v2
 
+#define GetCollaterals oe_get_collaterals
+
 #define VerifyReport oe_verify_report
+#define VerifyReportWithCollaterals oe_verify_report_with_collaterals
```

Comment:
Why do you need PascalCase versions at all?

Reply:
Opened #2205 for this.